src/azul/project/hca/transformers.py

```diff
@@ -23,26 +23,56 @@ def _project_dict(bundle: api.Bundle) -> dict:
     }
 
 
-def _specimen_dict(biomaterials: Mapping[api.UUID4, api.Biomaterial]) -> List[Mapping[str, Any]]:
```

Comment:
If we drop the assertion, this would be equivalent to `[_specimen_dict(s) for s in specimens.values()]`

I think should remove the method and inline the above fragment at the sole call site. The assertion is nice but I'd rather have one less method, and not have to explain the specimens/specimen subtlety to a reader of the source code.